src/system.c

```diff
@@ -535,10 +535,18 @@ int sys_notify_respond(int fd, struct seccomp_notif_resp *resp)
  */
 int sys_notify_id_valid(int fd, uint64_t id)
 {
+	int rc;
 	if (state.sup_user_notif <= 0)
 		return -EOPNOTSUPP;
 
-	if (ioctl(fd, SECCOMP_IOCTL_NOTIF_ID_VALID, &id) < 0)
+	rc = ioctl(fd, SECCOMP_IOCTL_NOTIF_ID_VALID, &id);
+	if (rc < 0 && errno == EINVAL)
+		/* We were built against newer kernel headers but are running on an
+		 * older kernel which doesn't accept the fixed ioctl number. Try the
+		 * old one.
+		 * See kernel commit 47e33c05f9f07cac3de833e531bcac9ae052c7ca */
```

Comment:
It might be nice to use the kernel standard of a 12-char commit ID followed by the subject line in parentesis and double quotes, e.g. `0123456789ab ("seccomp: let's break compatibility!")`.